adapters/repos/db/vector/hnsw/index.go

```diff
@@ -744,3 +717,97 @@ func IsHNSWIndex(index any) bool {
 	_, ok := index.(*hnsw)
 	return ok
 }
+
+type pair struct {
```

Comment:
I think the name of this type is too generic to be part of the hnsw package namespace.
How about something more specific, like 'nodeAddr', `nodeLevel` or anything else that conveys more meaning to the type?